core/src/test/scala/unit/kafka/raft/RaftManagerTest.scala

```
@@ -177,6 +201,78 @@ class RaftManagerTest {
     assertFalse(fileLocked(lockPath))
   }
 
+  @Test
+  def testMigratingZkBrokerDeletesMetadataLog(): Unit = {
+    val logDir = Some(TestUtils.tempDir().toPath)
+    val metadataDir = Some(TestUtils.tempDir().toPath)
+    val nodeId = 1
+    val config = createZkBrokerConfig(migrationEnabled = true, nodeId, logDir, metadataDir)
+    val raftManager = createRaftManager(
+      new TopicPartition("__cluster_metadata", 0),
+      config
+    )
+    raftManager.shutdown()
+
+    try {
+      KafkaRaftManager.maybeDeleteMetadataLogDir(config)
+      assertFalse(Files.exists(metadataDir.get.resolve("__cluster_metadata-0")))
+    } catch {
+      case err: Throwable => fail("Failed to delete metadata log", err)
```

it seems we can just throw the exception to fail this test case